src/engine/TPA.cc

```diff
@@ -1589,8 +1592,16 @@ VerificationResult TransitionSystemNetworkManager::solve() && {
     while (true) {
         if (getNode(current).blocked_children == getNode(current).children.size()) {
             if (current == graph.getEntry()) {
-                // TODO: Compute witness
-                return {VerificationAnswer::SAFE, NoWitness{"Unable to compute witness from TPA for DAG of TSs yet"}};
+                if (Options::COMPUTE_WITNESS == "true") {
```

Comment:
This is not going to work, `Options::COMPUTE_WITNESS` is just the string `"compute-witness"`, so this comparison is always `false`.  `TPAEngine` has `options` field, which you can query.
You will have to expose it somehow. I suggest to add a helper public method `bool shouldComputeWitness() const`. It can also be used internally in other places in TPA as well.

Reply:
Ok, thanks!